File: leetcode/0014. 最长公共前缀/main.go
package main

func main() {
	println(longestCommonPrefix([]string{"flower", "flow", "flight"}))
}

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	if len(strs) == 1 {
		return strs[0]
	}
	var prefix string
	for i := 0; i < len(strs[0]); i++ {
		for j := 0; j < len(strs); j++ {
			if i >= len(strs[j]) || strs[j][i] != strs[0][i] {
				return prefix
			}
		}
		prefix += string(strs[0][i])
	}
	return prefix
}

File: leetcode/0014. 最长公共前缀/main.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    if (strs.length === 0) return ''
    let res = strs[0]
    for (let i = 1; i < strs.length; i++) {
        while (strs[i].indexOf(res) !== 0) {
        res = res.substring(0, res.length - 1)
        }
    }
    return res
};

(function () {
    let strs = ['flower', 'flow', 'flight']
    console.log(longestCommonPrefix(strs))
})()